### PR TITLE
Ensure that HTTP OPTIONS handling involves the proper handler chain

### LINKED ISSUE #26828: RESTEasy Reactive does not execute response filters for OPTIONS requests
### Describe the bug

JAX-RS `ContainerResponseFilter`s as well as Quarkus-style `@ServerResponseFilter`s are not executed for OPTIONS method requests by RESTEasy Reactive. Works with RESTEasy Classic (the former only, as the latter is unavailable).

### Expected behavior

They should be executed.

### Actual behavior

They aren't.

### How to Reproduce?

Please see the demo project: [crf-test.tar.gz](https://github.com/quarkusio/quarkus/files/9145550/crf-test.tar.gz)

1. Run `./gradlew test`. All tests will pass.
2. In `build.gradle.kts`, replace `io.quarkus:quarkus-resteasy` with `io.quarkus:quarkus-resteasy-reactive`.
3. Run `./gradlew test` again. The OPTIONS test won't pass.

Note that there is no Quarkus-style `@ServerResponseFilter` test, because it's unavailable for RESTEasy classic. But it's affected, too.

### Output of `uname -a` or `ver`

Linux ***** 5.18.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 15 Jul 2022 15:33:02 +0000 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.15" 2022-04-19 OpenJDK Runtime Environment GraalVM CE 22.1.0 (build 11.0.15+10-jvmci-22.1-b06) OpenJDK 64-Bit Server VM GraalVM CE 22.1.0 (build 11.0.15+10-jvmci-22.1-b06, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.10.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.4.2

### Additional information

This bug hinders implementing custom CORS filters.

### PR DESCRIPTION
**Pull request by geoand**
This is done in order to be more consistent with how
other cases are handled and how RESTEasy Classic handles things as well

Fixes: #26828

### CODE REVIEW

**Comment by gsmet** [comment]
@glefloch these two Gradle tests fail very often. It would be good if you could have a look at the failures.
@Sgitario this RESTEasy Reactive test is failing quite often. I think it was you're doing? Could you have a look at what's going on?

**Comment by geoand** [comment]
@FroMage can you take a look at this? Should be an easy review :)

Thanks

**Comment by geoand** [comment]
@FroMage can you have a look at this please?

Thanks

**Comment by gsmet** [comment]
@FroMage ping ^

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/OptionsRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @Test
+    public void testJsonHeaderAdded() {
+        String allowValue = when()
+                .options("/test")
+                .then()
+                .statusCode(200)
+                .header("Foo", equalTo("Bar"))
+                .extract().header("Allow");
+
+        Assertions.assertThat(allowValue).contains("GET", "HEAD", "OPTIONS");
```
Comment (Sgitario) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/OptionsRequestTest.java:43]: For completeness, I would assert that the body is not "hello" to ensure the endpoint is not invoked.

**Review by Sgitario — APPROVED** [review]
lgtm, just a minor thing.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/OptionsRequestTest.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/OptionsRequestTest.java:46]: Very good point, added

**Pull request merged**